File: Principles_Of_Programming_Languages/pascal/largest_of_three.pas
program largest_of_three;
(*to find the largest number out of three*)

var
  a, b, c : integer;
begin
  writeln('Enter 3 numbers to find the largest of them: ');
  readln(a, b, c);

  if ((a > b) and (a > c)) then
    writeln('The largest is: ', a)
  else if ((b > a) and (b > c)) then
    writeln('The largest is: ', b)
  else
    writeln('The largest is: ', c);
end.

(*
Output example:

Enter 3 numbers to find the largest of them:
3
2
4
The largest is: 4

*)

File: Principles_Of_Programming_Languages/prolog/largest_of_three.pl
%Find_the_Largest_of_Three_Numbers

largest_of_three(A, B, C, Largest) :-
    (A >= B, A >= C -> Largest = A ;
     B >= A, B >= C -> Largest = B ;
     Largest = C).

find_Largest :-
    writeln('Enter three numbers:'), nl,
    read(X),
    read(Y),
    read(Z), nl,
    largest_of_three(X, Y, Z, Largest),
    write('The largest number is '), writeln(Largest), nl.

% Output example:
% ?- find_Largest.
% Enter three numbers:
% |: 45.
% |: 33.
% |: 99.
% 
% The largest number is 99